server/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -81,16 +81,20 @@ void innerUpdatePipelines(ClusterState previousState, ClusterState state) {
         }
 
         Map<String, Pipeline> pipelines = new HashMap<>();
+        ArrayList<ElasticsearchParseException> exceptions = new ArrayList<>();
         for (PipelineConfiguration pipeline : ingestMetadata.getPipelines().values()) {
             try {
                 pipelines.put(pipeline.getId(), factory.create(pipeline.getId(), pipeline.getConfigAsMap(), processorFactories));
             } catch (ElasticsearchParseException e) {
-                throw e;
+                pipelines.put(pipeline.getId(), Pipeline.EMPTY);
```

Comment:
I wonder if we can create a pipeline with a single `fail` processor. This fail processor can then contain the reason why the pipeline in question couldn't be loaded. This way there is no need to check the logs.

Reply:
I like this, but I will try to do this in such a way that the pipeline isn't actually executed and is only used to store the exception. If I can't find a way to cleanly do that, then I will just let the pipeline run and fail

Reply:
ah, also, `fail` processor is inside `ingest-common` so that makes it slightly more annoying